submissions/QuickPerformance/QuickPerformance.md

```diff
@@ -0,0 +1,113 @@
+## Quick Performance Check <author>oof2win2</author>
+
+![Image of a dropped connection](dropped.png)
+{ % include image.html src="dropped.png" alt="Image of the player being dropped from the game" }
+
+If you don't have a state-of-the-art computer, you probably have seen a message like this saying that you were dropped from the game because your network or computer was not fast enough. Well, what does that actually mean? What can you do about it in the short and long term?
+
+### Network Speed Issues
+
+Sometimes, your network may not be the best. That can be due to many things itself, such as maintenance from your ISP or a power outage of your DNS's datacenters. There are a few utilities that can however help mitigate these sorts of issues.
+
+To explain how Factorio's networking works, that is a longer process. However, *cube* has explained this process nicely in [FFF#136](https://factorio.com/blog/post/fff-136). One of the largest things they have done is using UDP but adding back things that are in TCP. [UDP](https://en.wikipedia.org/wiki/User_Datagram_Protocol) has the advantage of there being much faster speed than conventional TCP (which is used for most day-to-day things). UDP achieves this speed by not having a "handshake", which is two clients (computers) agreeing that they want to communicate together and establishing a connection, which simply means it doesn't care about what happens with the data, it *just sends it*. This however also removes the ability to guarantee the delivery of data, ordering the data properly or removing data duplicates (which can sometimes happen). 
+
+The fact that UDP removes the guarantee of delivering data however directly contradicts how Factorio must work using fully-deterministic, lockstep algorithms as I explained in an [earlier post](https://alt-f4.blog/ALTF4-26/), as the data *must* arrive or you would recieve a desync. *cube* said that Wube have solved this with implementing parts of the TCP protocol into their version of UDP. This means that they still have the speed that they want, but they can also have the reliability of TCP (or close to it, as the Factorio protocol is not publicly available for people to analyze).
+
+UDP is also very important because it allows NAT punching to work. NAT punching is the process of your client asking an external server (Pingpong server) to tell the Factorio server to establish a connection with your client, because some routers/firewalls may block "random" connections from IPs that you have not requested anything from, which is good in situations like preventing someone from brute-forcing open your computer, but an obstacle for the Factorio developers.
```

Comment:
> Pingpong

Maybe state what this is and/or link to a description of it?

Reply:
Not sure if this is good enough. What do you think?

NAT punching is the process of your client asking an external server ([Pingpong server](https://www.factorio.com/blog/post/fff-143), which is a server hosted by Factorio that serves just for this purpose)